python/langsmith/_internal/_operations.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import annotations
+
+import itertools
+import uuid
+from dataclasses import dataclass
+from typing import Literal, Optional, Union, cast
+
+import orjson
+
+from langsmith import schemas as ls_schemas
+from langsmith._internal._multipart import MultipartPart, MultipartPartsAndContext
+from langsmith._internal._serde import dumps_json as _dumps_json
+
+
+@dataclass
+class SerializedRunOperation:
```

Comment:
Do we actually need these to be dataclasses?

Reply:
discuss in other one